src/core/services.ts

```diff
@@ -1,76 +1,88 @@
-              null,
-              new Uint8Array(details.requestBody.raw[0].bytes) as any
-            )
+export const registerChromeEvents = () => (
+  send: Sender<CoreEvents>,
+  onReceive: any
+) => {
+  function onBeforeRequest(details: any) {
+    if (details.requestBody.raw[0]) {
+      const queryDetails = JSON.parse(
+        decodeURIComponent(
+          String.fromCharCode.apply(
+            null,
+            new Uint8Array(details.requestBody.raw[0].bytes) as any
           )
-        );
+        )
+      );
 
-        send({
-          type: 'ON_REQUEST',
-          payload: {
-            request: {
-              requestId: details.requestId,
-              query: queryDetails.query,
-              variables: queryDetails.variables
-            }
-          }
-        });
-      }
-    },
-    { urls: ['<all_urls>'] },
-    ['requestBody']
-  );
-
-  //@ts-ignore
-  chrome.webRequest.onCompleted.addListener(
-    function (details: any) {
       send({
-        type: 'ON_REQUEST_COMPLETE',
+        type: 'ON_REQUEST',
         payload: {
-          requestId: details.requestId,
-          statusCode: details.statusCode
-        }
+          request: {
+            requestId: details.requestId,
+            query: queryDetails.query,
+            variables: queryDetails.variables,
+          },
+        },
       });
-    },
-    { urls: ['<all_urls>'] }
-  );
+    }
+  }
 
-  //@ts-ignore
-  chrome.webRequest.onErrorOccurred.addListener(
-    function (details: any) {
-      send({
-        type: 'ON_REQUEST_ERROR',
-        payload: {
-          requestId: details.requestId,
-          statusCode: details.error === canceled_error ? 'canceled' : 'error'
-        }
-      });
-    },
-    { urls: ['<all_urls>'] }
-  );
+  function onCompleted(details: any) {
+    send({
+      type: 'ON_REQUEST_COMPLETE',
+      payload: {
+        requestId: details.requestId,
+        statusCode: details.statusCode,
+      },
+    });
+  }
 
-  //@ts-ignore
-  chrome.webRequest.onBeforeSendHeaders.addListener(
-    function (details: any) {
-      send({
-        type: 'ON_BEFORE_SEND_HEADERS',
-        payload: {
-          requestId: details.requestId,
-          requestHeaders: details.requestHeaders
-        }
-      });
-    },
-    { urls: ['<all_urls>'] },
-    ['requestHeaders']
-  );
+  function onErrorOccurred(details: any) {
+    const canceled_error = 'net::ERR_ABORTED';
+
+    send({
+      type: 'ON_REQUEST_ERROR',
+      payload: {
+        requestId: details.requestId,
+        statusCode: details.error === canceled_error ? 'canceled' : 'error',
+      },
+    });
+  }
+
+  function onBeforeSendHeaders(details: any) {
+    send({
+      type: 'ON_BEFORE_SEND_HEADERS',
+      payload: {
+        requestId: details.requestId,
+        requestHeaders: details.requestHeaders,
+      },
+    });
+  }
+
+  function registerAllEvents(urls: string[]) {
+    Object.values(chromeListeners).map((event) =>
+      event({
```

Comment:
if you wonder why I'm passing the functions to each event, what happens is that in order to make the `removeListerner` work correctly, it needs to be called passing the same callback reference.  the method `registerChromeEvents ` is called just once, while the register events could be called multiple times now, it depends on the user usage.